common/lib/xmodule/xmodule/conditional_module.py

```diff
@@ -227,65 +300,31 @@ def get_icon_class(self):
         # OBSOLETE: This obsoletes 'type'
         class_priority = ['video', 'problem']
 
-        child_classes = [self.system.get_module(child_descriptor).get_icon_class()
-                         for child_descriptor in self.descriptor.get_children()]
+        child_classes = [
+            child_descriptor.get_icon_class() for child_descriptor in self.get_children()
+        ]
         for c in class_priority:
             if c in child_classes:
                 new_class = c
         return new_class
 
-    def validate(self):
-        """
-        Message for either error or warning validation message/s.
-
-        Returns message and type. Priority given to error type message.
-        """
-        return self.descriptor.validate()
-
-
-class ConditionalDescriptor(ConditionalFields, SequenceDescriptor, StudioEditableDescriptor):
-    """Descriptor for conditional xmodule."""
-    _tag_name = 'conditional'
-
-    module_class = ConditionalModule
-
-    resources_dir = None
-
-    filename_extension = "xml"
-
-    has_score = False
-
-    show_in_read_only_mode = True
-
-    def __init__(self, *args, **kwargs):
-        """
-        Create an instance of the conditional module.
-        """
-        super(ConditionalDescriptor, self).__init__(*args, **kwargs)
-
-        # Convert sources xml_attribute to a ReferenceList field type so Location/Locator
-        # substitution can be done.
-        if not self.sources_list:
-            if 'sources' in self.xml_attributes and isinstance(self.xml_attributes['sources'], six.string_types):
-                self.sources_list = [
-                    # TODO: it is not clear why we are replacing the run here (which actually is a no-op
-                    # for old-style course locators. However, this is the implementation of
-                    # CourseLocator.make_usage_key_from_deprecated_string, which was previously
-                    # being called in this location.
-                    BlockUsageLocator.from_string(item).replace(run=self.location.course_key.run)
-                    for item in ConditionalDescriptor.parse_sources(self.xml_attributes)
-                ]
-
     @staticmethod
     def parse_sources(xml_element):
         """ Parse xml_element 'sources' attr and return a list of location strings. """
         sources = xml_element.get('sources')
         if sources:
             return [location.strip() for location in sources.split(';')]
 
+    @lazy
+    def get_required_blocks(self):
+        """
+        Returns a list of bound XBlocks instances upon which XBlock depends.
+        """
+        return [self.system.get_module(descriptor) for descriptor in self.get_required_module_descriptors()]
+
     def get_required_module_descriptors(self):
```

Comment:
We need to keep this because it is called by LMS to determine all the dependent XBlocks for which student state needed to be loaded. If we remove this then the XBlock instance will have empty student state after `bind_for_student()`.

https://github.com/edx/edx-platform/blob/98af9ce4185bde0a38030923a88403a101e011ab/lms/djangoapps/courseware/module_render.py#L641
https://github.com/edx/edx-platform/blob/98af9ce4185bde0a38030923a88403a101e011ab/common/lib/xmodule/xmodule/x_module.py#L575